hnn_core/gui/_viz_manager.py

```diff
@@ -0,0 +1,575 @@
+
+        self.axes_config_output = Output()
+        self.figs_output = Output()
+
+        # widgets
+        self.axes_config_tabs = Tab()
+        self.figs_tabs = Tab()
+        self.axes_config_tabs.selected_index = None
+        self.figs_tabs.selected_index = None
+        link(
+            (self.axes_config_tabs, 'selected_index'),
+            (self.figs_tabs, 'selected_index'),
+        )
+
+        template_names = list(fig_templates.keys())
+        self.templates_dropdown = Dropdown(
+            description='Layout template:',
+            options=template_names,
+            value=template_names[0],
+            style={'description_width': 'initial'},
+            layout=Layout(width="98%"))
+        self.make_fig_button = Button(
+            description='Make figure',
+            button_style="primary",
+            style={'button_color': self.viz_layout['theme_color']},
+            layout=self.viz_layout['btn'])
+        self.make_fig_button.on_click(self.add_figure)
+
+        # data
+        self.fig_idx = {"idx": 1}
+        self.figs = {}
+        self.gui_data = gui_data
+
+    @property
+    def widgets(self):
+        return {
+            "figs_output": self.figs_output,
+            "axes_config_tabs": self.axes_config_tabs,
+            "figs_tabs": self.figs_tabs,
+            "templates_dropdown": self.templates_dropdown
+        }
+
+    @property
+    def data(self):
+        return {
+            "use_ipympl": self.use_ipympl,
+            "simulations": self.gui_data["simulation_data"],
+            "fig_idx": self.fig_idx,
+            "visualization_output": self.viz_layout['visualization_output'],
+            "figs": self.figs
+        }
+
+    def reset_fig_config_tabs(self):
+        simulation_names = tuple(self.data['simulations'].keys())
+        for tab in self.axes_config_tabs.children:
+            controls = tab.children[1]
+            for ax_control in controls.children:
+                simulation_selection = ax_control.children[0]
+                simulation_selection.options = simulation_names
+
+    def compose(self):
+        with self.axes_config_output:
+            display(self.axes_config_tabs)
+        with self.figs_output:
+            display(Label(_fig_placeholder))
+
+        fig_output_container = VBox(
+            [self.figs_output], layout=self.viz_layout['visualization_window'])
+
+        config_panel = VBox([
+            Box(
+                [
+                    self.templates_dropdown,
+                    self.make_fig_button,
+                ],
+                layout=Layout(
+                    display='flex',
+                    flex_flow='column',
+                    align_items='stretch',
+                ),
+            ),
+            Label("Figure config:"),
+            self.axes_config_output,
+        ])
+        return config_panel, fig_output_container
+
+    def add_figure(self, b=None):
```

Comment:
I will keep repeating like a parrot ... public functions must have docstrings. It's non-negotiable :)